### PR TITLE
[Installer] Verify no duplicate static libs

### PR DESCRIPTION
**Pull request by DanToml**
Closes https://github.com/CocoaPods/CocoaPods/issues/4014

### CODE REVIEW

File: lib/cocoapods/installer.rb [hunk 1]
```diff
@@ -399,6 +400,21 @@ def verify_no_duplicate_framework_names
       end
     end
 
+    def verify_no_duplicate_library_names
```
Comment (manuyavuz) [inline lib/cocoapods/installer.rb]: @DanToml what about adding this check into `verify_no_duplicate_framework_names` and rename it `verify_no_duplicate_framework_and_library_names`? Or moving common logic into a base method and use it inside these two methods? Currently these two functions are almost the same, and could be handled together in a method, or use a common base method.

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Samuel Giddins](https://github.com/segiddins)
   [#5321](https://github.com/CocoaPods/CocoaPods/issues/5321)
 
+* Error during install when there are duplicate library names.
```
Comment (manuyavuz) [inline CHANGELOG.md]: Missing trailing two spaces.

File: lib/cocoapods/installer.rb (hunk 1 above)
Comment (DanToml) [inline lib/cocoapods/installer.rb]: Good question. I don't generally like to add layers of abstraction for fewer than three uses (especially on code that doesn't change often), but we could do something along the lines of:

```ruby
def verify_no_duplicate_frameworks_and_libraries_names
  aggregate_targets.each do |aggregate_target|
    aggregate_target.user_build_configurations.keys.each do |config|
      pod_targets = aggregate_target.pod_targets_for_build_configuration(config)
      verify_no_duplicate_framework_names_in_pod_targets(targets)
      verify_no_duplicate_library_names_in_pod_targets(targets)
    end
  end
end

def verify_no_duplicate_framework_names_in_pod_targets(targets)
  vendored_frameworks = pod_targets.flat_map(&:file_accessors).flat_map(&:vendored_frameworks).uniq
  frameworks = vendored_frameworks.map { |fw| fw.basename('.framework') }
  frameworks += pod_targets.select { |pt| pt.should_build? && pt.requires_frameworks? }.map(&:product_module_name)

  duplicates = frameworks.group_by { |f| f }.select { |_, v| v.size > 1 }.keys
  unless duplicates.empty?
    raise Informative, "The '#{aggregate_target.label}' target has " \
      "frameworks with conflicting names: #{duplicates.to_sentence}."
  end
end

def verify_no_duplicate_library_names_in_pod_targets(targets)
  libraries = pod_targets.flat_map(&:file_accessors).flat_map(&:vendored_libraries).uniq.map(&:basename)
  libraries += pod_targets.select { |pt| pt.should_build? && !pt.requires_frameworks? }.map(&:product_name)
  duplicates = libraries.group_by { |l| l.to_s.downcase }.select { |_, v| v.size > 1 }.keys
  unless duplicates.empty?
    raise Informative, "The '#{aggregate_target.label}' target has " \
      "libraries with conflicting names: #{duplicates.to_sentence}."
  end
end
```

but I'm not sure that we actually get too much from it

File: CHANGELOG.md (hunk 2 above)
Comment (DanToml) [inline CHANGELOG.md]: Oops, was stripped by vim. Will fix now.

**Comment by segiddins** [comment]
👍🏻

File: spec/unit/installer_spec.rb [hunk 3]
```diff
@@ -305,6 +305,28 @@ def @installer.analyze(*)
 
     #-------------------------------------------------------------------------#
 
+    describe '#verify_no_duplicate_library_names' do
```
Comment (mrackwitz) [inline spec/unit/installer_spec.rb]: This should either refer to an actual existing method or be put into natural case.

**Comment by mrackwitz** [comment]
@DanToml: Could you rebase this branch and address my comment above, so that we can get this merged and :ship:ed?

**Comment by DanToml** [comment]
@mrackwitz Yeah 👍, will get it sorted this evening.

**Comment by orta** [comment]
This'll need nother rebase - sorry

**Comment by DanToml** [comment]
@orta Done, should be good to :shipit: when CI is done.

**Pull request merged**